bin/validate-templates.sh

```diff
@@ -109,14 +119,18 @@ function validate_templates() {
   return 0
 }
 
-if [ -n "$1" ]; then
-  [ -n "$VERBOSE" ] && echo "Running validate-templates for target cluster only"
-  validate_templates
-  # re-enable next line after helm does not throw error any more: https://github.com/helm/helm/issues/8596
-  # hf lint
-else
-  [ -n "$VERBOSE" ] && echo "Running validate-templates for all clusters"
-  for_each_cluster validate_templates
-  # re-enable next line after helm does not throw error any more: https://github.com/helm/helm/issues/8596
-  # for_each_cluster hf lint
+function main() {
+  parse_args "$@"
+  [[ $all && $label ]] && echo "Error: cannot specify --all and --label simultaneously" && exit 6
+  if [[ $all = 'y' || $label ]]; then
+    for_each_cluster validate_templates
+    exit 0
+  fi
+}
+
+if [[ "${BASH_SOURCE[0]}" == "${0}" ]]; then
+  main "$@"
+  if [ $? -gt 0 ]; then
+    exit 1
+  fi
```

Comment:
Main plumbing.

There are 127 exit codes and we only use 2. I deliberately used some to `assert_failure $exit_code`. Correct me if I'm wrong.

Reply:
ok, but is that consistent? Confuses me to assign numbers without meaning. Where are these codes documented?

Reply:
I don't want to suggest it is a panacea or perfect solution, no.

I provided exit codes documentation in the function comment for `parse_args`. I can imagine that if someone checks for the exit code of the script it might be confusing. So I don't know what a good solution there would be.

But for `bats` tests it is quite nice, as one can `assert_failure $exit_code`.

Reply:
Here is another idea for standardisation: https://stackoverflow.com/a/24121322.

Reply:
I think I have a better compromise. I improved the `err` function to be a bit more verbose.